discovery/src/server.rs

```diff
@@ -12,7 +12,7 @@ use aes::cipher::{KeyIvInit, StreamCipher};
 use base64::engine::general_purpose::STANDARD as BASE64;
 use base64::engine::Engine as _;
 use futures_core::Stream;
-use futures_util::{FutureExt, TryFutureExt};
+use futures_util::{lock::Mutex, FutureExt, TryFutureExt};
```

Comment:
Please use `Mutex` from `std` instead of `async` muteness.

Reply:
Done.